pkg/apis/operator/v1alpha1/tektonhub_default_test.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSetDefault(t *testing.T) {
+
+	th := &TektonHub{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hub",
+		},
+		Spec: TektonHubSpec{
+			Api: ApiSpec{
+				HubConfigUrl:  "https://hubconfig.url",
```

Comment:
Why do we have defaulted this value for hub config?

Reply:
We haven't defaulted the `HubConfigUrl` value. This is the required field hence we need to specify this in struct for the validation webhook to validate this.